### PR TITLE
Configure and install version file when building in Trilinos

### PR DESCRIPTION
**Pull request by aprokop**
This is a fix to install version file to address the missing part in #2779.

### CODE REVIEW

**Comment by dalg24** [comment]
OK to test

File: cmake/kokkos_install.cmake [hunk 1]
```diff
@@ -35,6 +35,13 @@ ELSE()
   CONFIGURE_FILE(cmake/KokkosTrilinosConfig.cmake.in ${Kokkos_BINARY_DIR}/KokkosTrilinosConfig.cmake @ONLY)
   file(READ ${Kokkos_BINARY_DIR}/KokkosTrilinosConfig.cmake KOKKOS_TRILINOS_CONFIG)
   file(APPEND "${CMAKE_CURRENT_BINARY_DIR}/CMakeFiles/KokkosConfig_install.cmake" "${KOKKOS_TRILINOS_CONFIG}")
+
+  WRITE_BASIC_PACKAGE_VERSION_FILE("${CMAKE_CURRENT_BINARY_DIR}/KokkosConfigVersion.cmake"
+      VERSION "${Kokkos_VERSION}"
+      COMPATIBILITY SameMajorVersion)
+
+  install(FILES ${CMAKE_CURRENT_BINARY_DIR}/KokkosConfigVersion.cmake
+      DESTINATION "${${PROJECT_NAME}_INSTALL_LIB_DIR}/cmake/${PACKAGE_NAME}")
```
Comment (dalg24) [inline cmake/kokkos_install.cmake:44]: @bartlettroscoe please confirm this is the right way to deduce the destination

**Comment by bartlettroscoe** [comment]
When building inside of Trilinos, the various `<Package>Config.cmake` files get generated automatically.  There should be no code in Kokkos itself that does this.

NOTE: We will be updating what these `<Package>Config.cmake` files look like as part of https://github.com/TriBITSPub/TriBITS/issues/63 and https://github.com/TriBITSPub/TriBITS/issues/299.  The plan is to devote a lot of time starting in May to that work.

We are also working on a standard for these `<Package>Config.cmake` files in:

* https://gitlab.kitware.com/snl/project-1/-/issues/139

**Comment by crtrott** [comment]
@bartlettroscoe this is not doing anything with KokkosConfig.cmake it only adds a Version.cmake file

**Comment by dalg24** [comment]
> We are also working on a standard for these `<Package>Config.cmake` files in:
> 
> * https://gitlab.kitware.com/snl/project-1/-/issues/139

I don't have the right permissions to see that page

**Comment by bartlettroscoe** [comment]
> I don't have the right permissions to see that page

**Comment by bartlettroscoe** [comment]
> I don't have the right permissions to see that page

@dalg24, can you contact me offline and I will see about getting you access to that site?

**Comment by bartlettroscoe** [comment]
> this is not doing anything with KokkosConfig.cmake it only adds a Version.cmake file

@crtrott, okay.  But as part of addressing https://github.com/TriBITSPub/TriBITS/issues/63, we may need to write a specialized `<Package>ConfigVersion.cmake` file to get that system to work (to differentiate standard compatible packages from non-standard compatible packages).

File: cmake/kokkos_install.cmake (hunk 1 above)
Comment (bartlettroscoe) [reply cmake/kokkos_install.cmake:44]: This looks like the correct location. 

NOTE: As I mentioned [here](https://github.com/kokkos/kokkos/pull/2957#issuecomment-614690130), TriBITS may need to create this file as well in order to prefer finding "standard compliant" packages.

**Comment by jjwilke** [comment]
This looks fine as a fix for now.

**Comment by sethrj** [comment]
Another argument for Trilinos as a TPL (see https://github.com/TriBITSPub/TriBITS/issues/63) and kokkos as a TPL as well. SCALE/VERA/MPACT (for example) can't use on any projects that require also require Trilinos or Kokkos.

**Pull request merged**